cpp/benchmarks/common/generate_benchmark_input.cpp

```diff
@@ -347,14 +347,22 @@ void copy_string(cudf::size_type src_idx,
 template <typename Char_gen>
 void append_string(Char_gen& char_gen, bool valid, uint32_t length, string_column_data& column_data)
 {
-  auto const idx = column_data.offsets.size() - 1;
-  column_data.offsets.push_back(column_data.offsets.back() + length);
-  std::generate_n(std::back_inserter(column_data.chars),
-                  column_data.offsets[idx + 1] - column_data.offsets[idx],
-                  [&]() { return char_gen(); });
-
-  // TODO: use empty string for invalid fields?
-  if (!valid) { cudf::clear_bit_unsafe(column_data.null_mask.data(), idx); }
+  if (!valid) {
+    auto const idx = column_data.offsets.size() - 1;
+    cudf::clear_bit_unsafe(column_data.null_mask.data(), idx);
+    // duplicate the offset value to indicate an empty row
+    column_data.offsets.push_back(column_data.offsets.back());
+    return;
+  }
+  std::generate_n(std::back_inserter(column_data.chars), length, [&]() mutable {
+    auto ch = char_gen();
+    if (ch < '\x7F') return static_cast<char>(ch);
+    // x7F is at the top edge of ASCII;
+    // the next set of characters are assigned two bytes
+    column_data.chars.push_back('\xC4');
```

Comment:
See comment from dependent PR: https://github.com/rapidsai/cudf/pull/7316#discussion_r570571169

Reply:
Consensus from C++ meetup was to change this to a for-loop.
https://github.com/rapidsai/cudf/pull/7316#discussion_r570698106